poetry/masonry/builders/sdist.py

```diff
@@ -55,16 +55,19 @@ def build(self, target_dir=None):  # type: (Path) -> Path
         if not target_dir.exists():
             target_dir.mkdir(parents=True)
 
-        target = target_dir / "{}-{}.tar.gz".format(
-            self._package.pretty_name, self._meta.version
+        basename = (
+            self._package.pretty_name
+            if self._package.distribution is None
+            else self._package.distribution
         )
```

Comment:
you can shorten this expression to: `basename = self._package.distribution or self._package.pretty_name`. also apllies for the `wheels` module in this package.